code/main.py

```diff
@@ -0,0 +1,285 @@
+        '''
+            Initialize Featurizer object based on the user selected data
+
+            Note:
+                Selection required for: filename_input, filename_output, featurization type
+        '''
+        featurizer = Featurizer.Featurizer.from_file(
+            self.filename_input, self.specification, use_numpy=self.use_numpy.get(),
+            verbose=self.verbose.get()
+        )
+        verbose_text = featurizer.get_features_from_classes_helper()
+        return featurizer, verbose_text
+
+    def button_browse_input(self):
+        ''' Browse Button for filename_input '''
+        button = ttk.Button(self.labelFrame, text="Browse", command=self.fileDialog)
+        button.grid(column=1, row=0)
+
+    def fileDialog(self):
+        '''
+            command associated with Browse Button for filename_input
+            Allowed filetype: .txt
+        '''
+        self.filename_input = filedialog.askopenfilename(initialdir="./", title="Select a file",
+                                                         filetype=(("TXT", "*.txt"), ("All Files", "*.*")))
+        label_input = ttk.Label(self.labelFrame, text="")
+        label_input.grid(column=2, row=0)
+        label_input.configure(text=self.filename_input)
+
+    def button_browse_output_PNG_feat(self):
+        ''' Browse button to output PNG (feat) file based on the selected options '''
+        button = ttk.Button(self.labelFrame5, text="Featurization graph as PNG", command= lambda: self.fileDialogSave_PNG(FEAT))
+        button.grid(column=1, row=0, sticky='ew')
+
+    def button_browse_output_PNG_poset(self):
+        ''' Browse button to output PNG (poset) file based on the selected options '''
+        button = ttk.Button(self.labelFrame5, text="Poset graph as PNG", command= lambda: self.fileDialogSave_PNG(POSET))
+        button.grid(column=1, row=1, sticky='ew')
+
+    def fileDialogSave_PNG(self, output_type):
+        '''
+            command associated with Browse Button for output_PNG
+            Allowed filetype: .png
+        '''
+        if not self.input_validation():
+            return
+
+        filename_default = self.generate_default_filename() + "_" + output_type[1]
+        filename_output = filedialog.asksaveasfilename(
+            confirmoverwrite=True,
+            filetypes=[("PNG", "*.png")],
+            defaultextension='.png',
+            initialdir="./",
+            initialfile=filename_default,
+            title="Choose a file name"
+        )
+        # fileDialog cancelled
+        if not filename_output:
+            return
+
+        featurizer, *_ = self.init_featurizer()
+        filename_temp = filename_output.rsplit('/', 1)[0] + '/temp.temp'
+        if output_type == FEAT:
+            featurizer.graph_feats(filename_temp)
+        elif output_type == POSET:
+            featurizer.graph_poset(filename_temp)
+
+        try:
+            featurizer.dot_to_png(filename_temp, filename_output)
+        except:
```

Comment:
Can you switch this to catch the `graphviz.backend.ExecutableNotFound` error specifically? What you've got here is called 'Pokemon exception handling' (gotta catch 'em all). This is sometimes ok, but in this case it's dangerous because it will report a missing graphviz executable for EVERY exception, not just those that are triggered by the missing executable. If the issue was something like file permissions, this would be very misleading.

I'd also expand the first error message below a little bit to say something like "Missing graphviz executable" just to be a bit more concrete.